packages/data-point/lib/entity-types/entity-control/resolve.js

```diff
@@ -2,16 +2,16 @@ const Promise = require('bluebird')
 
 /**
  *
- * @param {any} caseStatements
- * @param {any} acc
- * @param {any} resolveReducer
- * @returns
+ * @param {Array<Object>} caseStatements
+ * @param {Accumulator} accumulator
+ * @param {Function} resolveReducer
+ * @return {Number}
  */
-function getMatchingCaseStatement (caseStatements, acc, resolveReducer) {
+function getMatchingCaseIndex (caseStatements, accumulator, resolveReducer) {
```

Comment:
returning the index instead of the statement, so that we can add the index to the reducer stack